components/hab/src/cli/hab/sup.rs

```diff
@@ -338,6 +338,21 @@ pub struct SupRun {
     /// process (default: set in plan)
     #[structopt(name = "SHUTDOWN_TIMEOUT", long = "shutdown-timeout")]
     shutdown_timeout: Option<ShutdownTimeout>,
+    /// Automatically cleanup old package versions.
+    ///
+    /// If this flag is enabled, service startup will initiate an uninstall of all previous
+    /// versions of the associated package. This also applies when a service is restarted due to an
+    /// update. If a number is passed to this argument, that number of latest versions will be
+    /// kept. The same logic applies to the Supervisor package if self updates are enabled.
+    // TODO (DM): We want this to have a default value of 1 if set otherwise it should be none. Does
+    // Option<Option<T>> achieve this
+    #[structopt(name = "KEEP_LATEST_PACKAGES",
+                long = "keep-latest-packages",
+                default_value = "1",
+                max_values = 2, // This is needed to use an env var
```

Comment:
:thinking: That's unexpected :thinking:

Reply:
This was a hack. Fixing the issue discussed [here](https://github.com/habitat-sh/habitat/pull/7582#discussion_r401095494). Addressed in [1036bc](https://github.com/habitat-sh/habitat/pull/7582/commits/1036bc25017ea5df6a92a7bd6e8537e4ce175b03).